test/unit/controllers/AppFooterControllerSpec.js

```diff
@@ -2,34 +2,35 @@
 /* global describe, it, inject, expect, beforeEach */
 
 describe('AppFooterController', function () {
-  var $controller, $scope, service, serviceFlag
-
   beforeEach(module('myApp.controllers'))
 
   beforeEach(function () {
-    serviceFlag = false
-    service = {
+    this.LayoutSwitchService = {
+      serviceFlag: false,
       switchLayout: function (parameter) {
-        serviceFlag = true
+        this.serviceFlag = true
       }
     }
 
+    var LayoutSwitchService = this.LayoutSwitchService
+
     inject(function (_$controller_, _$rootScope_) {
-      $controller = _$controller_
+      this.$controller = _$controller_
 
-      $scope = _$rootScope_.$new()
-      $controller('AppFooterController', {
+      this.$scope = _$rootScope_.$new()
+      var $scope = this.$scope
+      this.$controller('AppFooterController', {
         $scope: $scope,
```

Comment:
Ah, here. Hm, but `this.$scope` should work, too. You're passing pointers to object after all...